### ISSUE TITLE
Pyright reporting variable not used even if it is defined beginning with _

### ISSUE THREAD

**Issue by bellini666** [microsoft/pyright#2497]
Hi,

Pieces of code like this gives a lot of warnings in the editor:

```python
for _i in range(10):  # it warns me that "_i" is not used
    print("foobar")
```

I know that if I change that to, for example:

```python
for _ in range(10):
    print("foobar")
```

Pyright will not give me the warnings anymore.

But I can't do that in my projects because we use `gettext`, and it is usually mapped to `_` so we can write messages like `_("Some Message")` and `gettext` will collect them for translation.

I don't want to disable the warnings because I find the "variable not used" useful, but I want to have a way to tell pyright that, "yes, I know that it is not used but that's ok".

That happens also when overloading some func where I need to match the exactly signature, even if I won't use all the params (but that maybe is another conversation)

**Comment by MarcelRobitaille** [comment]
@bellini666 Did you find a way to gray out the text in neovim?

**Comment by bellini666** [comment]
@MarcelRobitaille yeah, I monkey patched the LSP's client `"textDocument/publishDiagnostics"` to filter out `HINT` from the diagnostics. You can check if in my dotfiles in this line: https://github.com/bellini666/dotfiles/blob/master/vim/lua/config/lsp.lua#L31

Note that I'm filtering everything lower than `INFO` there (which in this case, are `HINTS`) because I don't even want them to be populated to my fix list when I want to list all diagnostic errors in the project. If you still want those to be used elsewhere but want to remove it from one or all of underline/signs/virtualtext/etc, you can do a per-option configuration like this: https://github.com/neovim/nvim-lspconfig/wiki/UI-customization#customizing-how-diagnostics-are-displayed . You can pass a table instead of `true/false` for more advanced options, like what I'm also doing there to set a minimum severity of `WARN` to display the underline (so that `INFO` doesn't get underlined, even though it still displays signs and virtualtext)

Hope that helps! :)

**Comment by MarcelRobitaille** [comment]
@bellini666 I see. I'm working on something similar. You didn't find a way to gray out the text as well then?

**Comment by bellini666** [comment]
> @bellini666 I see. I'm working on something similar. You didn't find a way to gray out the text as well then?

Wait, I think I misunderstood your question then... You want to not only underline the text but also to gray out it in the background? If so, I think you may want to override the highlight for `DiagnosticUnderlineHint`. But I might still be misunderstanding what you want =P

**Comment by MarcelRobitaille** [comment]
I was wondering if it's possible to get Pyright's intended behaviour in neovim:

> Pyright doesn't emit warnings for unaccessed variables. It does emit a "hint" that tell the editor that it should display the text in a grayed-out manner as a subtle hint that the symbol is not accessed.

I monkey-patched `on_publish_diagnostics` in order to remove the virtual text for unused variables if they begin with `_`, but I was wondering if you were also able to gray the text. Now that I read the conversation again, I realize that it was not you who mentioned graying the text.
